[CLOUDCRAFT-354] Version History (#24367)
DataDog/documentation: content/en/cloudcraft/getting-started/version-history.md

---
title: Version History
---

## Overview

Version history allows users to track changes in their architecture diagrams over time, allowing them to review and restore previous iterations of diagrams. Whether you're managing complex cloud architectures or collaborating with a team, version history provides invaluable insight into your diagram's evolution.

You can access the version history feature by clicking the **Version history** button in the top right corner of the Cloudcraft application.

{{< img src="cloudcraft/getting-started/version-history/cloudcraft-diagram-aws-infrastructure-version-history.png" alt="Cloudcraft diagram showcasing AWS infrastructure with an arrow highlighting the version history button." responsive="true" style="width:100%;">}}

## Working with versions

Version history isn't just for reviewing past work—it's also a powerful tool for managing your current and future diagrams. Here are some key actions you can take:

1. **Restoring previous versions**: If you need to revert to a previous state of your diagram, you can easily do so. Simply click the **Restore this version** button on the top right side of your screen when viewing a past version.

> 2. **Creating new blueprints**: Version history allows you to save any specific version of your diagram as a new blueprint. This feature is particularly useful for creating templates or preserving specific architectural states for future reference. To create a new blueprint from a version, click the select the three dots on the right side of the version name and choose **Save as a new blueprint**.

Reviewer: ```suggestion
2. **Creating new blueprints**: Version history allows you to save any specific version of your diagram as a new blueprint. This feature is particularly useful for creating templates or preserving specific architectural states for future reference. To create a new blueprint from a version, click the three dots on the right side of the version name and choose **Save as a new blueprint**.
```

3. **Comparing versions**: While not explicit, the ability to view different versions allows for manual comparison of how your architecture has changed over time.

> Note that currently you cannot manually create or delete versions from your history.

Reviewer: ```suggestion
**Note**: Manually creating or deleting versions from your history is not available.
```
For this, we to avoid words like "currently" or future based functionality

### Creation

Versions are automatically created as you work on your diagrams. By default, each version is timestamped and named after the date and time of its creation. However, for easier reference, you have the option to give specific versions custom names, which is particularly useful for marking significant milestones or changes in your architecture.

To name a version:

1. Select the version you want to name.
2. Click the three dots on the right side of the version name.
3. Choose **Name this version** from the dropdown menu.

{{< img src="cloudcraft/getting-started/version-history/version-history-interface-cloudcraft.png" alt="Version history interface with options to name or save versions in Cloudcraft." responsive="true" style="width:100%;">}}

Cloudcraft creates new versions intelligently to balance granularity with efficiency. If the current version is more than 5 minutes old, any new updates will trigger the creation of a new version. For changes made within 5 minutes of the latest version, updates are added to that existing version. This approach ensures that your version history remains meaningful without becoming cluttered with minor changes.

> It's worth noting that while you cannot manually create a new version at present, a new version is automatically generated when you switch from [Snapshot to Live mode][1] in your diagram.

Reviewer: ```suggestion
It's worth noting that while you cannot manually create a new version, one is automatically generated when you switch from [Snapshot to Live mode][1] in your diagram.
```
